autodiff/forward/forward.hpp

```diff
@@ -458,6 +458,8 @@ struct Dual
 
     Dual() : Dual(0.0) {}
 
+    operator T() const { return this->val; }
```

Comment:
Thanks for the PR. I think an `explicit` conversion is required:

~~~c++
explicit operator T() const { return this->val; }
~~~

to avoid "accidental" conversions to `T` by the compiler, and thus spoiling the propagation of derivatives.

Let's see if the tests pass.

Reply:
Thanks, that seems sensible.

Reply:
I think it would be more conservative if we go `explicit` here. I'll make this change.

Reply:
Could you please check if using `explicit` conversion works fine with Eigen on your side?

Reply:
It seems to be working ok on some test cases. 
I can't run the full tests in ./autodiff/test. I've only been using the headers from the library thus far.

Reply:
OK. What I actually meant is that if the issue #26 is resolved with `explicit` conversion operator, instead of implicit. I cannot test it right now. Actually, it would make sense to have a test there. Can you give me a simple example, please, that shows the casting combined with Eigen usage?

Reply:
I've added two tests to test/forward.test.cpp showing vector and matrix casts.
They pass locally (seems to have passed Travis too).